decompiler/pipeline/controlflowanalysis/variable_name_generation.py

```diff
@@ -108,13 +115,15 @@ def __init__(self, task: DecompilerTask) -> None:
     def renameVariableNames(self):
         """Rename all collected variables to the hungarian notation."""
         for var in self._variables:
+            if self.alread_renamed(var._name):
+                continue
             counter = _get_var_counter(var.name)
             var._name = self._hungarian_notation(var, counter if counter else "")
```

Comment:
Could you ellaborate on this change?
As far as I can see, you assume `self._variables` contains variables already renamed, but I fail to pinpoint how this can come to pass.

Reply:
Ah thank you for the explaination! I didn't think about duplicate references in the list.
However, could you not fix this for example by habing `self._variables` be a set? afaik, the hash method of variables should be implemented like this:
`return hash(repr(self))`

If not, I think we should document this state somewhere in the file to avoid confusion when someone else tries to work on it.

Reply:
I already tried the set approach, but exactly because of our hash implementation is does not work. 

For example: Lets say we have 3 variables, all with the same type etc., where two are the same object and the other one don't. 

All objects have the same hash, because of `repr(self)`, but only one of them in the set.
Therefore only the two same, or the single one will be renamed, yielding at least one unnamed variable. 

I'll add a small documentation at the top of the file (after includes).